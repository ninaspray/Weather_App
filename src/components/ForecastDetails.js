import React from "react";
import moment from "moment";
import ForecastSummary from "./ForecastSummary";
import PropTypes from "prop-types;"

const ForecastDetails = ({ forecast }) => {
   const { date, humidity, temperature, wind } = forecast;

   return (
       <div className="forcast-details">
           <p className="forecast-details-date">
               {moment(date).format("ddd Do MMM")}
           </p>
           <p className="forecast-detail_temp">
               <b>Max Temperature:</b>
               {temperature.max}
               &deg;C
           </p>
           <p className="forecast-detail_temp">
               <b>Min Temperature</b>
               {temperature.min}
               &deg;C
           </p>
           <p className="forecast-details_humidity">
                <b>Humidity:</b>
                {`${humidity}%`}
           </p>
           <p className="forecast-details_wind">
                <b>Wind:</b>
                {`${wind.speed}mph`}
           </p>
       </div>
   );

};

export default ForecastDetails;

ForecastDetails.propTypes = {
    forecast: PropTypes.shape({
        date: PropTypes.number,
        humidity: PropTypes.number,
        temperature: PropTypes.shape({
            min: PropTypes.number,
            max: PropTypes.number,
          }),
          wind: PropTypes.shape({
            speed: PropTypes.number,
            direction: PropTypes.string,
          }),
        }).isRequired,
      };