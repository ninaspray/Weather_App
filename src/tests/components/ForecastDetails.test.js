import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("Forecast Details", () => {
  const validProps = {
    date: 123456,
    humidity: 30,
    temperature: {
      min: 2,
      max: 52,
    },
    wind: {
        speed: 5,
        direction: "w",
    }
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });
});