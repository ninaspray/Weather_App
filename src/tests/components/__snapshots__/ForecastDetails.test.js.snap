// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forecast Details renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forcast-details"
  >
    <p
      class="forecast-details_date"
    >
      Thu 1st Jan
    </p>
    <p
      class="forecast-detail_temp"
    >
      <b>
        Max Temperature:
      </b>
      52°C
    </p>
    <p
      class="forecast-detail_temp"
    >
      <b>
        Min Temperature
      </b>
      2°C
    </p>
    <p
      class="forecast-details_humidity"
    >
      <b>
        Humidity:
      </b>
      30%
    </p>
    <p
      class="forecast-details_wind"
    >
      <b>
        Wind:
      </b>
      5mph
    </p>
  </div>
</DocumentFragment>
`;
